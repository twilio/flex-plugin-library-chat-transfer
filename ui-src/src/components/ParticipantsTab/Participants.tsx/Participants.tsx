import { Stack, Card, Heading } from '@twilio-paste/core';

import { Participant } from './Participant/Participant';
import { ParticipantDetails } from '../../../flex-hooks/types/ParticipantDetails';
import React from 'react';
interface ParticipantsProps {
  participantDetails: ParticipantDetails[];
  handleKickParticipant: (interactionParticipantSid: string) => void;
}

export const Participants = ({ participantDetails, handleKickParticipant }: ParticipantsProps) => {
  const participants = participantDetails.map((participantDetail) => {
    const { participantType } = participantDetail;
    const { isMe } = participantDetail;
    const allowKick = !isMe && participantType === 'agent';
    const participantName = participantDetail.friendlyName;
    const { interactionParticipantSid } = participantDetail;

    return (
      <Participant
        name={participantName}
        participantType={participantType}
        allowKick={allowKick}
        key={`participant-${interactionParticipantSid}`}
        handleKickParticiant={() => handleKickParticipant(interactionParticipantSid)}
      />
    );
  });

  return (
    <Card padding="space60">
      <Heading as="h2" variant="heading20"> </Heading>
      <Stack orientation="vertical" spacing="space20">
        {participants}
      </Stack>
    </Card>
  );
};
