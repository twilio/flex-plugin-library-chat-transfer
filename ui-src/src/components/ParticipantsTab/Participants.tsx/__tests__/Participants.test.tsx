import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Participants } from '../Participants';

jest.mock('@twilio/flex-ui', () => ({
  Template: jest.fn(({ source }) => <span>{source}</span>),
  templates: {
    [Symbol()]: 'MockedTemplate',
  },
}));

describe('Participants', () => {
  const mockParticipantDetails = [
    { friendlyName: 'Participant 1', participantType: 'agent', isMe: false, interactionParticipantSid: '123' },
    { friendlyName: 'Participant 2', participantType: 'customer', isMe: true, interactionParticipantSid: '456' },
  ];
  it('renders participants with correct details', () => {
    const mockHandleKickParticipant = jest.fn();

    const { getByText } = render(
      <Participants participantDetails={mockParticipantDetails} handleKickParticipant={mockHandleKickParticipant} />
    );
    expect(getByText).toMatchSnapshot();
  });
});
