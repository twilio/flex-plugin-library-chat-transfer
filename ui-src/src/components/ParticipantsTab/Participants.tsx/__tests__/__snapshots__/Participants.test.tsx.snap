// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participants renders participants with correct details 1`] = `[Function]`;
